packages/plugins/src/foundry/utils/index.ts

```diff
@@ -1176,18 +1097,163 @@ export const findFullyQualifiedNameForInitCode = (
  */
 export const findFullyQualifiedNameForAddress = (
   address: string,
-  inputs: Array<RawActionInput>,
+  accountAccesses: Array<AccountAccess>,
   configArtifacts: ConfigArtifacts
 ): string | undefined => {
-  address
-  inputs
-  configArtifacts
-  return undefined
+  const createAccountAccess = accountAccesses.find(
+    (accountAccess) =>
+      accountAccess.kind === AccountAccessKind.Create &&
+      accountAccess.account === address
+  )
+
+  if (createAccountAccess) {
+    const initCodeWithArgs = createAccountAccess.data
+    return findFullyQualifiedNameForInitCode(initCodeWithArgs, configArtifacts)
+  } else {
+    return undefined
+  }
+}
+
+/**
+ * Write the ABI encoded Sphinx system contracts to the file system. This is useful when we need to
+ * pass the system contract info from TypeScript to a Forge script. We write it to the file system
+ * instead of passing it in as a parameter to the Forge script because it's possible to hit Node's
+ * `spawn` input size limit if the data is large. This is particularly a concern because the data
+ * contains contract init code.
+ */
+export const writeSystemContracts = (
+  sphinxPluginTypesInterface: ethers.Interface,
+  cachePath: string
+): string => {
+  const systemContractsArrayFragment = findFunctionFragment(
+    sphinxPluginTypesInterface,
+    'systemContractInfoArrayType'
+  )
+
+  const coder = ethers.AbiCoder.defaultAbiCoder()
+  const filePath = join(cachePath, 'sphinx-system-contracts.txt')
+
+  const encodedSystemContractArray = coder.encode(
+    systemContractsArrayFragment.outputs,
+    [getSystemContractInfo()]
+  )
+
+  // Write the ABI encoded data to the file system. We'll read it in the Forge script. We do this
+  // instead of passing in the data as a parameter to the Forge script because it's possible to hit
+  // Node's `spawn` input size limit if the data is large. This is particularly a concern because
+  // the data contains contract init code.
+  writeFileSync(filePath, encodedSystemContractArray)
+
+  return filePath
+}
+
+export const assertValidAccountAccesses = (
+  accountAccesses: Array<AccountAccess>,
+  safeAddress: string
+): void => {
+  for (const accountAccess of accountAccesses) {
+    if (accountAccess.accessor === safeAddress) {
+      if (BigInt(accountAccess.value) > BigInt(0)) {
+        throw new Error(
+          `Detected value transfer in script:\n` +
+            `To: ${accountAccess.account}\n` +
+            `Amount: ${ethers.parseEther(accountAccess.value.toString())}\n` +
+            `Sphinx does not support transferring native gas tokens during deployments.\n` +
+            `Let us know if you want this feature!`
+        )
+      }
```

Comment:
Why isn't this allowed? I don't see any reason why we can't support sending value from the safe to some other address as long as the safe is funded. I thought the limitation in the readme was wrt to airdropping tokens to an unfunded safe.

Reply:
You're right - fixed in [this PR](https://github.com/sphinx-labs/sphinx/pull/1415/files)